### ISSUE TITLE
lib/feature.t: blocker for strict-on-default

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#228]
As of commit ac6b7ccf73 in the alpha-dev-02-strict branch on Aug 18 2020, the file cited is failing.  Here is output from `cd t; ./perl harness -v <file>; cd -`:
```
# ../lib/feature.t

PROG: 
# No say; should be a syntax error.
use warnings;
say "Hello", "world";
EXPECTED:
Unquoted string "say" may clash with future reserved word at - line 3.
String found where operator expected at - line 3, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 3, near "say "Hello""
Execution of - aborted due to compilation errors.
GOT:
String found where operator expected at - line 3, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 3, near "say "Hello""
Execution of - aborted due to compilation errors.
# Failed test 43 - at lib/feature/say line 5
PROG: 
# feature out of scope; should be a syntax error.
use warnings;
{ use feature 'say'; }
say "Hello", "world";
EXPECTED:
Unquoted string "say" may clash with future reserved word at - line 4.
String found where operator expected at - line 4, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 4, near "say "Hello""
Execution of - aborted due to compilation errors.
GOT:
String found where operator expected at - line 4, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 4, near "say "Hello""
Execution of - aborted due to compilation errors.
# Failed test 46 - at lib/feature/say line 29
PROG: 
# 'no feature' should work
use warnings;
use feature 'say';
say "Hello", "world";
no feature;
say "Hello", "world";
EXPECTED:
Unquoted string "say" may clash with future reserved word at - line 6.
String found where operator expected at - line 6, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 6, near "say "Hello""
Execution of - aborted due to compilation errors.
GOT:
String found where operator expected at - line 6, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 6, near "say "Hello""
Execution of - aborted due to compilation errors.
# Failed test 47 - at lib/feature/say line 40
PROG: 
# 'no feature "say"' should work too
use warnings;
use feature 'say';
say "Hello", "world";
no feature 'say';
say "Hello", "world";
EXPECTED:
Unquoted string "say" may clash with future reserved word at - line 6.
String found where operator expected at - line 6, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 6, near "say "Hello""
Execution of - aborted due to compilation errors.
GOT:
String found where operator expected at - line 6, near "say "Hello""
	(Do you need to predeclare say?)
syntax error at - line 6, near "say "Hello""
Execution of - aborted due to compilation errors.
# Failed test 48 - at lib/feature/say line 53
PROG: 
# No switch; given should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
print STDOUT given;
EXPECTED:
Unquoted string "given" may clash with future reserved word at - line 3.
given
GOT:
Bareword "given" not allowed while "strict subs" in use at - line 3.
Execution of - aborted due to compilation errors.
# Failed test 49 - at lib/feature/switch line 5
PROG: 
# No switch; when should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
print STDOUT when;
EXPECTED:
Unquoted string "when" may clash with future reserved word at - line 3.
when
GOT:
Bareword "when" not allowed while "strict subs" in use at - line 3.
Execution of - aborted due to compilation errors.
# Failed test 50 - at lib/feature/switch line 12
PROG: 
# No switch; default should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
print STDOUT default;
EXPECTED:
Unquoted string "default" may clash with future reserved word at - line 3.
default
GOT:
Bareword "default" not allowed while "strict subs" in use at - line 3.
Execution of - aborted due to compilation errors.
# Failed test 51 - at lib/feature/switch line 19
PROG: 
# No switch; break should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
print STDOUT break;
EXPECTED:
Unquoted string "break" may clash with future reserved word at - line 3.
break
GOT:
Bareword "break" not allowed while "strict subs" in use at - line 3.
Execution of - aborted due to compilation errors.
# Failed test 52 - at lib/feature/switch line 26
PROG: 
# switch out of scope; given should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
{ use feature 'switch';
  given (1) {print "Okay here\n";}
}
print STDOUT given;
EXPECTED:
Unquoted string "given" may clash with future reserved word at - line 6.
Okay here
given
GOT:
Bareword "given" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 58 - at lib/feature/switch line 62
PROG: 
# switch out of scope; when should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
{ use feature 'switch';
  given (1) { when(1) {print "Okay here\n";} }
}
print STDOUT when;
EXPECTED:
Unquoted string "when" may clash with future reserved word at - line 6.
Okay here
when
GOT:
Bareword "when" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 59 - at lib/feature/switch line 73
PROG: 
# switch out of scope; default should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
{ use feature 'switch';
  given (1) { default {print "Okay here\n";} }
}
print STDOUT default;
EXPECTED:
Unquoted string "default" may clash with future reserved word at - line 6.
Okay here
default
GOT:
Bareword "default" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 60 - at lib/feature/switch line 84
PROG: 
# switch out of scope; break should be a bareword.
use warnings; no warnings 'experimental::smartmatch';
{ use feature 'switch';
  given (1) { break }  
}
print STDOUT break;
EXPECTED:
Unquoted string "break" may clash with future reserved word at - line 6.
break
GOT:
Bareword "break" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 61 - at lib/feature/switch line 95
PROG: 
# C<no feature 'switch'> should work
use warnings; no warnings 'experimental::smartmatch';
use feature 'switch';
given (1) { when(1) {print "Okay here\n";} }
no feature 'switch';
print STDOUT when;
EXPECTED:
Unquoted string "when" may clash with future reserved word at - line 6.
Okay here
when
GOT:
Bareword "when" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 62 - at lib/feature/switch line 105
PROG: 
# C<no feature> should work too
use warnings; no warnings 'experimental::smartmatch';
use feature 'switch';
given (1) { when(1) {print "Okay here\n";} }
no feature;
print STDOUT when;
EXPECTED:
Unquoted string "when" may clash with future reserved word at - line 6.
Okay here
when
GOT:
Bareword "when" not allowed while "strict subs" in use at - line 6.
Execution of - aborted due to compilation errors.
# Failed test 63 - at lib/feature/switch line 116
../lib/feature.t .. 
1..65
# From lib/feature/bits
ok 1 - check clearing $^H clears the bits
...
# From lib/feature/say
not ok 43
ok 44
ok 45
not ok 46
not ok 47
not ok 48
# From lib/feature/switch
not ok 49
not ok 50
not ok 51
not ok 52
ok 53
ok 54
ok 55
ok 56
ok 57
not ok 58
not ok 59
not ok 60
not ok 61
not ok 62
not ok 63
ok 64
ok 65
Failed 14/65 subtests 

Test Summary Report
-------------------
../lib/feature.t (Wstat: 0 Tests: 65 Failed: 14)
  Failed tests:  43, 46-52, 58-63
Files=1, Tests=65,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.35 cusr  0.11 csys =  0.49 CPU)
Result: FAIL

```

@atoomic @toddr 

Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
this is now fixed by d7a0f2b5

**Comment by jkeenan** [comment]
> this is now fixed by [d7a0f2b](https://github.com/atoomic/perl/commit/d7a0f2b52148dfa1be66deefff9e46f99a570ff4)

Confirmed.
```
$ extract-sha
627eedcc50
$ cd t; TEST_JOBS=1 ./perl harness ../lib/overload.t ../lib/feature.t; cd -
../lib/overload.t .. ok         
../lib/feature.t ... ok     
All tests successful.
Files=2, Tests=5428,  1 wallclock secs ( 0.19 usr  0.00 sys +  0.57 cusr  0.09 csys =  0.85 CPU)
Result: PASS
```
Closing.

Thank you very much.
Jim Keenan
